src/tests/lib/utils/tooltip.utils.spec.ts

```diff
@@ -0,0 +1,170 @@
+import { translateTooltip } from "$lib/utils/tooltip.utils";
+
+describe("translateTooltip", () => {
+  const containerRect = {
+    x: 100,
+    y: 200,
+    width: 400,
+    height: 600,
+    top: 200,
+    left: 100,
+    bottom: 800,
+    right: 500,
+  };
+
+  const targetWidth = 50;
+  const targetHeight = 30;
+
+  const tooltipWidth = 200;
+  const tooltipHeight = 40;
+
+  const createTargetAndTooltipRects = ({ x, y }) => ({
+    targetRect: {
+      x,
+      y,
+      width: targetWidth,
+      height: targetHeight,
+      top: y,
+      left: x,
+      bottom: y + targetHeight,
+      right: x + targetWidth,
+    },
+    tooltipRect: {
+      x,
+      y,
+      width: tooltipWidth,
+      height: tooltipHeight,
+      top: y,
+      left: x,
+      bottom: y + tooltipHeight,
+      right: x + tooltipWidth,
+    },
+  });
+
+  describe("when the target isn't close to any edge", () => {
+    const { targetRect, tooltipRect } = createTargetAndTooltipRects({
+      x: containerRect.x + 200,
+      y: containerRect.y + 200,
+    });
+    const scrollbarWidth = 0;
+
+    it("should position the tooltip centered below the target", () => {
+      expect(
+        translateTooltip({
+          containerRect,
+          targetRect,
+          tooltipRect,
+          scrollbarWidth,
+        }),
+      ).toEqual({
+        x: (targetRect.width - tooltipRect.width) / 2,
+        y: targetRect.height,
+      });
+    });
+
+    it("should position the tooltip centered above the target when 'top' is true", () => {
+      expect(
+        translateTooltip({
+          containerRect,
+          targetRect,
+          tooltipRect,
+          scrollbarWidth,
+          top: true,
+        }),
+      ).toEqual({
+        x: (targetRect.width - tooltipRect.width) / 2,
+        y: -tooltipRect.height,
+      });
+    });
+  });
+
+  describe("when the target is close to the left edge", () => {
+    const { targetRect, tooltipRect } = createTargetAndTooltipRects({
+      x: containerRect.x + 10,
+      y: containerRect.y + 200,
+    });
+    const scrollbarWidth = 0;
+
+    it("should position the tooltip touching the left edge", () => {
+      expect(
+        translateTooltip({
+          containerRect,
+          targetRect,
+          tooltipRect,
+          scrollbarWidth,
+        }),
+      ).toEqual({
+        x: -10,
```

Comment:
Is this `-10` because you positioned it `+ 10` to the `containerRect.x`?

If so, could we use a variable wiht the amount?

Reply:
Done